packages/core/core/src/services/document-service/transform/relations/transform/data-ids.ts

```diff
@@ -1,29 +1,60 @@
 import { EntityService, Attribute, Common } from '@strapi/types';
-import { errors, traverseEntity } from '@strapi/utils';
+import { traverseEntity } from '@strapi/utils';
 import { isObject, isNil } from 'lodash/fp';
-import { ShortHand, LongHand, ID, GetIdOrThrow } from '../utils/types';
+import { ShortHand, LongHand, ID, GetId } from '../utils/types';
 import { isShortHand, isLongHand } from '../utils/data';
 import { IdMap } from '../../id-map';
 
+const isNumeric = (value: any): value is number => {
+  const parsed = parseInt(value, 10);
+  return !Number.isNaN(parsed);
+};
+
 const transformPrimitive = <T extends ShortHand | LongHand>(
   relation: T | T[] | null | undefined,
-  getId: GetIdOrThrow
+  getId: GetId
 ): T | T[] | undefined => {
+  // If id value is a number, return it as is, it's already an entry id
+  if (isNumeric(relation)) {
```

Comment:
Weren't we going to remove this and require longhand for db ids?

Reply:
yep, thats the plan! But we are going to do that when we switch id to documentId everywhere, that will make the transition easier. For now I didnt want to add more noise into this pr if that's ok!